Source/Core/CesiumTerrainProvider.js

```diff
@@ -109,8 +110,16 @@ define([
         }
         //>>includeEnd('debug');
 
-        this._url = options.url;
-        this._proxy = options.proxy;
+        if (defined(options.proxy)) {
+            deprecationWarning('CesiumTerrainProvider.proxy', 'The options.proxy parameter has been deprecated. Specify options.url as a Resource instance and set the proxy property there.');
+        }
+
+        var resource = Resource.createIfNeeded(options.url, {
+            proxy: options.proxy
+        });
+        resource.isDirectory = true;
```

Comment:
Since we're modifying something the user passed in, should we clone this first?  The larger issue here is that terrain requires `isDirectory` to be true, but now that value has to be manually set if the url doesn't end in `/`?  Would a better option be to have `createDerivedResource` take an option like `isDirectory:true` instead of having it be a property of the Resource itself?

@shunter thoughts?

P.S. URIs are not file paths, so `isDirectory` is not a good name, perhaps `isBaseUri` or something similar?